OptimizelySDK/OptimizelyUserContext.cs

```diff
@@ -37,39 +37,49 @@ public class OptimizelyUserContext
 
         // user attributes for Optimizely user context.
         private UserAttributes Attributes;
-        
-        private readonly List<string> QualifiedSegments;
+
+        // set of qualified segments
+        public List<string> QualifiedSegments { get; }
 
         // Optimizely object to be used.
         private Optimizely Optimizely;
 
         private ForcedDecisionsStore ForcedDecisionsStore { get; set; }
 
-        public OptimizelyUserContext(Optimizely optimizely, string userId, UserAttributes userAttributes, IErrorHandler errorHandler, ILogger logger) :
-            this(optimizely, userId, userAttributes, null, errorHandler, logger)
-        {
-        }
+        public OptimizelyUserContext(Optimizely optimizely, string userId,
+            UserAttributes userAttributes, IErrorHandler errorHandler, ILogger logger
+        ) :
+            this(optimizely, userId, userAttributes, null, errorHandler, logger) { }
 
-        public OptimizelyUserContext(Optimizely optimizely, string userId, UserAttributes userAttributes, ForcedDecisionsStore forcedDecisionsStore, IErrorHandler errorHandler, ILogger logger)
+        public OptimizelyUserContext(Optimizely optimizely, string userId,
+            UserAttributes userAttributes, ForcedDecisionsStore forcedDecisionsStore,
+            IErrorHandler errorHandler, ILogger logger, List<string> qualifiedSegments = null
+        )
         {
             ErrorHandler = errorHandler;
             Logger = logger;
             Optimizely = optimizely;
             Attributes = userAttributes ?? new UserAttributes();
             ForcedDecisionsStore = forcedDecisionsStore ?? new ForcedDecisionsStore();
             UserId = userId;
-            QualifiedSegments = new List<string>();
+            QualifiedSegments = qualifiedSegments ?? new List<string>();
         }
 
-        private OptimizelyUserContext Copy() => new OptimizelyUserContext(Optimizely, UserId, GetAttributes(), GetForcedDecisionsStore(), ErrorHandler, Logger);
-        
+        private OptimizelyUserContext Copy() =>
+            new OptimizelyUserContext(Optimizely, UserId, GetAttributes(),
+                GetForcedDecisionsStore(), ErrorHandler, Logger, QualifiedSegments);
```

Comment:
```suggestion
                GetForcedDecisionsStore(), ErrorHandler, Logger, GetQualifiedSegments());
```
Should pass a copy (thread-safe) of qualifiedSegments.

Reply:
I was hoping to use .NET's [Auto-Implemented Properties](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/auto-implemented-properties), but if we need to ensure thread safety, I'll adjust and put in a private prop and thread-safe getter & setter with the mutex obj.